### PR TITLE
Feat/detected plate client implementation

### PR DESCRIPTION
**Pull request by DonatasTamosauskas**
Had a few moments on the plane - here are the results: Implemented DetectedPlateClient and created it's interface. Since this is work in the frontend, I can't test it, so please launch the app before merging.

### CODE REVIEW

File: HucaresWF/src/Hucares.Server.Client/DetectedPlateClient.cs [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+using System.Text;
+using System.Threading.Tasks;
+using Hucares.Server.Client.Models;
+using Newtonsoft.Json;
+
+namespace Hucares.Server.Client
+{
+    public class DetectedPlateClient : IDetectedPlateClient
+    {
+
```
Comment (SSMellow) [inline HucaresWF/src/Hucares.Server.Client/DetectedPlateClient.cs]: unnecessary line

File: HucaresWF/src/Hucares.Server.Client/IDetectedPlateClient.cs [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    public interface IDetectedPlateClient
+    {
+        Task<IEnumerable<DetectedLicensePlate>> GetAllDetectedMissingPlates();
+
+        Task<IEnumerable<DetectedLicensePlate>> GetAllDetectedPlatesByPlateNumber(string plateNumber,
+            DateTime? startDateTime = null, DateTime? endDateTime = null);
+
+        Task<IEnumerable<DetectedLicensePlate>> GetAllDetectedPlatesByCamera(int cameraId, DateTime? startDateTime = null, 
+            DateTime? endDateTime = null);
+
```
Comment (SSMellow) [inline HucaresWF/src/Hucares.Server.Client/IDetectedPlateClient.cs]: unnecessary lines

File: HucaresWF/src/Hucares.Server.Client/DetectedPlateClient.cs [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        public DetectedPlateClient(HttpClientHelper httpHelper = null)
+        {
+            this.httpHelper = httpHelper ?? new HttpClientHelper();
+        }
+        
+        public async Task<IEnumerable<DetectedLicensePlate>> GetAllDetectedMissingPlates()
+        {
+            var uri = "api/dlp/all";
+            var fullUri = new Uri(HostUri, uri);
+
```
Comment (SSMellow) [inline HucaresWF/src/Hucares.Server.Client/DetectedPlateClient.cs]: one line variables should not be seperated by new line

File: HucaresWF/src/Hucares.Server.Client/DetectedPlateClient.cs (hunk 1 above)
Comment (DonatasTamosauskas) [reply HucaresWF/src/Hucares.Server.Client/DetectedPlateClient.cs]: Removed

File: HucaresWF/src/Hucares.Server.Client/IDetectedPlateClient.cs (hunk 2 above)
Comment (DonatasTamosauskas) [reply HucaresWF/src/Hucares.Server.Client/IDetectedPlateClient.cs]: Destroyed

File: HucaresWF/src/Hucares.Server.Client/DetectedPlateClient.cs (hunk 3 above)
Comment (DonatasTamosauskas) [reply HucaresWF/src/Hucares.Server.Client/DetectedPlateClient.cs]: Divided and conquered

**Comment by DonatasTamosauskas** [comment]
Fixed issues, and added Ainoras' fix in regex error printout.

File: HucaresWF/src/HucaresWF/HucaresWF.csproj [hunk 4]
```diff
@@ -66,6 +70,7 @@
       <AutoGen>True</AutoGen>
       <DependentUpon>Resources.resx</DependentUpon>
     </Compile>
+    <None Include="packages.config" />
```
Comment (AinoraZ) [inline HucaresWF/src/HucaresWF/HucaresWF.csproj]: File added to compile but not included with commit

File: HucaresWF/src/Hucares.Server.Client/DetectedPlateClient.cs [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+using Hucares.Server.Client.Models;
+using Newtonsoft.Json;
+
+namespace Hucares.Server.Client
+{
+    public class DetectedPlateClient : IDetectedPlateClient
```
Comment (AinoraZ) [inline HucaresWF/src/Hucares.Server.Client/DetectedPlateClient.cs]: Class and interface not added to project compile

File: HucaresWF/src/HucaresWF/HucaresWF.csproj (hunk 4 above)
Comment (DonatasTamosauskas) [reply HucaresWF/src/HucaresWF/HucaresWF.csproj]: Added file to git

File: HucaresWF/src/Hucares.Server.Client/DetectedPlateClient.cs (hunk 5 above)
Comment (DonatasTamosauskas) [reply HucaresWF/src/Hucares.Server.Client/DetectedPlateClient.cs]: Added to .csproj

File: HucaresWF/src/Hucares.Server.Client/DetectedPlateClient.cs [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+                startDateTime = startDateTime,
+                endDateTime = endDateTime
+            };
+
+            var jsonContent = JsonConvert.SerializeObject(plateNumberData, new JsonSerializerSettings
+            {
+                NullValueHandling = NullValueHandling.Ignore
+            });
+            
+            var request = new HttpRequestMessage(HttpMethod.Post, fullUri)
```
Comment (AinoraZ) [inline HucaresWF/src/Hucares.Server.Client/DetectedPlateClient.cs]: Should be HttpMethod.Get

File: HucaresWF/src/Hucares.Server.Client/DetectedPlateClient.cs [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+            return await httpHelper.MakeRequest<IEnumerable<DetectedLicensePlate>>(request);
+        }
+
+        public async Task<IEnumerable<DetectedLicensePlate>> GetAllDetectedPlatesByPlateNumber(string plateNumber, DateTime? startDateTime = null,
+            DateTime? endDateTime = null)
+        {
+            var uri = $"api/dlp/plate/{plateNumber}";
+            var fullUri = new Uri(HostUri, uri);
+
+            var plateNumberData = new
```
Comment (AinoraZ) [inline HucaresWF/src/Hucares.Server.Client/DetectedPlateClient.cs]: Paramaters have to be passed via query syntax (for example: ?startDateTime=2017-12-31T22:00:00.000Z&endDateTime=2018-01-01T22:00:00.000Z)

File: HucaresWF/src/Hucares.Server.Client/DetectedPlateClient.cs [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+            return await httpHelper.MakeRequest<IEnumerable<DetectedLicensePlate>>(request);
+        }
+
+        public async Task<IEnumerable<DetectedLicensePlate>> GetAllDetectedPlatesByCamera(int cameraId, DateTime? startDateTime = null, 
+			DateTime? endDateTime = null)
+        {
+            var uri = $"api/dlp/cam/{cameraId}";
+            var fullUri = new Uri(HostUri, uri);
+
+            var cameraIdData = new
```
Comment (AinoraZ) [inline HucaresWF/src/Hucares.Server.Client/DetectedPlateClient.cs]: Paramaters have to be passed via query syntax (for example: ?startDateTime=2017-12-31T22:00:00.000Z&endDateTime=2018-01-01T22:00:00.000Z)

File: HucaresWF/src/Hucares.Server.Client/DetectedPlateClient.cs [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+                startDateTime = startDateTime,
+                endDateTime = endDateTime
+            };
+
+            var jsonContent = JsonConvert.SerializeObject(cameraIdData, new JsonSerializerSettings
+            {
+                NullValueHandling = NullValueHandling.Ignore
+            });
+            
+            var request = new HttpRequestMessage(HttpMethod.Post, fullUri)
```
Comment (AinoraZ) [inline HucaresWF/src/Hucares.Server.Client/DetectedPlateClient.cs]: HttpMethod.Get

**Pull request merged**